### DISCUSSION TITLE
Gempy 3d software computer requirements

### DISCUSSION THREAD

**Discussion by malflink** [gempy-project/gempy#963] [Q&A]
Hi
I am new to Gempy and have been following the tutorials. I can do most things, but when it comes to 3d, I would like to know what graphics is required. My laptop has an NDIVIA GeForce GTX 965M. What graphics should it be? thanks mike

**Comment by javoha** [comment]
Hi @malflink,
as it is often the case with open source projects like gempy they rely on other open source packages for certain tasks. Gempy uses a package called pyvista (https://docs.pyvista.org/) in the backend for visualization. I am not aware of any hardware restrictions. you should absolutely be fine with your setup.
You do have do install pyvista separately though as described linked on the gempy installation page (https://www.gempy.org/installation). Just use
`pip install pyvista`
in the same environment you are running gempy in. The 3D plotting should work then. Depending on your setup executing `gpv.plot_3d(geo_model)` should open a new window containing a interactive 3d plot of the model. 
Let me know if it works or if you have any other problems with it.
Cheers,
Jan

**Comment by malflink** [comment]
thks
mike

**Comment by NilsChudalla** [comment]
A small addition to Jan's great explanation: The main computational load is carried by your CPU to solve for your model by default. GPU usage is important for AI methods for research grade applications with GemPy. The 3D objects are generally not that complex in terms of vertices and shading, that the GPU should become the bottle neck. If you encounter issues with the plotting regardless, we gladly will help you finding a solution.

**Comment by malflink** [comment]
Thks alot
Mike

**Comment by EternalDeiwos** [comment]
I think @malflink's main limitation is going to be memory. Most python libs are quite hungry and the alternative of using any kind of remote solution like a server with more memory is not simple/obvious to set up (see gempy-project/gempy#954).
